db_go18.go

```diff
@@ -37,5 +37,14 @@ func query(ctx context.Context, ex executor, query string, args ...interface{})
 }
 
 func begin(db *DB) (*sql.Tx, error) {
-	return db.DB.BeginTx(db.context, nil)
+	return db.DB.BeginTx(db.Context(), nil)
+}
+
+// BeginTx begins a transaction with the provided context and options.
+func (db *DB) BeginTx(ctx context.Context, opts *sql.TxOptions) (*Tx, error) {
```

Comment:
Move to `db.go` to keep pattern of that implementing the public API. As precedent, `db.go` already defines `(db *DB) Begin`, which delegates to `begin` in this file.

Reply:
The problem is `sql.TxOptions` isn't a thing before 1.8, so it won't compile if moved to `db.go`.

Reply:
Ah, that makes sense.